### PR TITLE
Personal Proxy Update

### PR DESCRIPTION
**Pull request by Risewill23**
Suggestion from #343

### CODE REVIEW

File: src/proxy.py [hunk 1]
```diff
@@ -103,13 +103,18 @@ def get_proxies(self, proxies):
     return proxies_list
 
 
-def get_random_proxy(self):
+def get_random_proxy(self, name):
```
Comment (thoroc) [inline src/proxy.py:106]: What type is name? `str`? Could you document that? and add a default value of `None` if that is the intend

File: main.py [hunk 2]
```diff
@@ -264,7 +265,7 @@ def get_board(self, access_token_in):
                                         requests.get(
                                             msg["data"]["name"],
                                             stream=True,
-                                            proxies=proxy.get_random_proxy(self),
+                                            proxies=proxy.get_random_proxy(self, None),
```
Comment (thoroc) [inline main.py:268]: Don't pass a `None` value if you don't intend to give any as first argument. If you intend to make it apparent that the `name` is `None` then do the following:
```python
proxies=proxy.get_random_proxy(self, name=None)
```

File: main.py [hunk 3]
```diff
@@ -113,7 +114,7 @@ def set_pixel_and_check_ratelimit(
             url,
             headers=headers,
             data=payload,
-            proxies=proxy.get_random_proxy(self),
+            proxies=proxy.get_random_proxy(self, None),
```
Comment (thoroc) [inline main.py:117]: See comment about `name` argument being `None` below

**Review by thoroc — CHANGES_REQUESTED** [review]

**Review by ixi2101 — APPROVED** [review]

**Comment by Risewill23** [comment]
The changes need reviewed too I guess

File: src/proxy.py [hunk 4]
```diff
@@ -110,14 +110,19 @@ def get_proxies(self, proxies):
         return proxies_list
     return proxies_list
 
-
-def get_random_proxy(self):
+#name is the username of the worker and is used for personal proxies
+def get_random_proxy(self, name):
```
Comment (ixi2101) [inline src/proxy.py:114]: Should default to None here

**Review by ixi2101 — CHANGES_REQUESTED** [review]
It looks like you've added the name=None   to the function calls instead of the definition

File: main.py [hunk 5]
```diff
@@ -290,7 +291,7 @@ def get_board(self, access_token_in):
                                         requests.get(
                                             msg["data"]["name"],
                                             stream=True,
-                                            proxies=proxy.get_random_proxy(self),
+                                            proxies=proxy.get_random_proxy(self, name=None),
```
Comment (ixi2101) [inline main.py:294]: This is going to always call get_random_proxy with name set to None. I don't think that;s what your intending to do

**Review by ixi2101 — CHANGES_REQUESTED** [review]

File: main.py (hunk 5 above)
Comment (Risewill23) [reply main.py:294]: That's for non worker calls so I did mean to put that there

Comment (ixi2101) [reply main.py:294]: Why are you making the function call if it is always going to return the same value? That line of code won't perform anything

Comment (ixi2101) [reply main.py:294]: my bad

**Comment by Risewill23** [comment]
Can you review/approve the change request now?

**Comment by ixi2101** [comment]
Have you tested it working?

**Comment by Risewill23** [comment]
yes like right after I made the last change

**Review by ixi2101 — APPROVED** [review]

**Comment by Risewill23** [comment]
thanks

**Comment by ixi2101** [comment]
Linting is broken dw about that

**Pull request merged**